### PR TITLE
Fix remaining memory leaks, .clang-format, fixes

### PR DESCRIPTION
**Pull request by sleaper**
I hopefully fixed the other memory leaks in #157 . I also added .clang-format. I don't really care about styling, but it would be nice to keep it unified throughout the whole project. Feel free to change it.

Currently, we dynamically allocate memory for `dirs` in the `load_files` function. What I think would be better is to scan the `/exercise` folder for all the files and malloc all the necessary memory at the **beginning** (with some kind of init function). Then we would have just one cleanup function in the end.

image

### CODE REVIEW

File: src/files.c [hunk 1]
```diff
@@ -82,27 +79,49 @@ bool is_marked_incompleted(char *file_contents, int file_contents_size) {
...
     return false;
 }
 
 void add_dir(FileCollection **dirs, File *files, int file_ct, int dir_idx) {
+
+    // Free dir if not null
+    if (dirs[dir_idx] != NULL) {
+        if (dirs[dir_idx]->files != NULL) {
+            for (size_t e = 0; e < dirs[dir_idx]->file_ct; e++) {
+                free(dirs[dir_idx]->files[e].file_path);
```
Comment (sleaper) [inline src/files.c:95]: I do not like this. It would be nice to move the allocation to the beginning. Then we could free it just at the end.

**Comment by danwritecode** [comment]
I'll review this tomorrow! thanks for helping on this one

**Comment by sleaper** [comment]
Also added data types exercise: Resolves #131

**Review by danwritecode — APPROVED** [review]
looks good, thanks for tackling those remaining memory leaks.

**Comment by danwritecode** [comment]
FYI, i rebased master into 'fix/memory-leaks' to eliminate the huge diff, looks like that created some conflicts in your branch.

**Comment by sleaper** [comment]
Finally, it should be okay. I am sorry for the mess. Git is difficult for me.

**Pull request merged**